flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java

```diff
@@ -36,52 +35,57 @@
 
 import static org.apache.flink.runtime.state.ChannelPersistenceITCase.getStreamFactoryFactory;
 import static org.apache.flink.util.CloseableIterator.ofElements;
-import static org.apache.flink.util.ExceptionUtils.findThrowable;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 /** {@link ChannelStateWriterImpl} lifecycle tests. */
-public class ChannelStateWriterImplTest {
+class ChannelStateWriterImplTest {
     private static final long CHECKPOINT_ID = 42L;
     private static final String TASK_NAME = "test";
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testAddEventBuffer() throws Exception {
+    @Test
+    void testAddEventBuffer() {
 
         NetworkBuffer dataBuf = getBuffer();
         NetworkBuffer eventBuf = getBuffer();
         eventBuf.setDataType(Buffer.DataType.EVENT_BUFFER);
-        try {
-            runWithSyncWorker(
-                    writer -> {
-                        callStart(writer);
-                        writer.addInputData(
-                                CHECKPOINT_ID,
-                                new InputChannelInfo(1, 1),
-                                1,
-                                ofElements(Buffer::recycleBuffer, eventBuf, dataBuf));
-                    });
-        } finally {
-            assertTrue(dataBuf.isRecycled());
-        }
+        assertThatThrownBy(
```

Comment:
we should make the `assertThatThrownBy` specific to the call that causes the issue.

Reply:
The code after update, it should be specific. Thanks for your great suggestion, totally agree with your suggestion.

```
    @Test
    void testAddEventBuffer() throws Exception {

        NetworkBuffer dataBuf = getBuffer();
        NetworkBuffer eventBuf = getBuffer();
        eventBuf.setDataType(Buffer.DataType.EVENT_BUFFER);

        runWithSyncWorker(
                (writer, worker) -> {
                    callStart(writer);
                    callAddInputData(writer, dataBuf);
                    worker.processAllRequests();

                    callAddInputData(writer, eventBuf);
                    assertThatThrownBy(worker::processAllRequests)
                            .isInstanceOf(IllegalArgumentException.class);
                });
        assertThat(dataBuf.isRecycled()).isTrue();
    }
```